updater/osv/osv_test.go

```diff
@@ -138,18 +141,383 @@ func TestParse(t *testing.T) {
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "0",
+								},
+								{
+									Fixed: "0.4.0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("0.0.0")),
+					Upper: claircore.FromSemver(semver.MustParse("0.4.0")),
+				},
+				FixedInVersion: "0.4.0",
+			},
+		},
+	},
+	{
+		name: "unfixed",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "go",
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "0",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("0.0.0")),
+					Upper: claircore.Version{
+						Kind: "semver",
+						V:    [10]int32{65535, 0, 0, 0, 0, 0, 0, 0, 0, 0},
+					},
+				},
+				FixedInVersion: "",
+			},
+		},
+	},
+	{
+		name: "two_affected",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "go",
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "0",
+								},
+								{
+									Fixed: "0.4.10",
+								},
+							},
+						},
+					},
+				},
+				{
+					Package: _package{
+						Ecosystem: "go",
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "0.5.0",
+								},
+								{
+									Fixed: "0.5.3",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		expectedVulns: []claircore.Vulnerability{
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("0.0.0")),
+					Upper: claircore.FromSemver(semver.MustParse("0.4.10")),
+				},
+				FixedInVersion: "0.4.10",
+			},
+			{
+				Name:    "test1",
+				Updater: "test",
+				Range: &claircore.Range{
+					Lower: claircore.FromSemver(semver.MustParse("0.5.0")),
+					Upper: claircore.FromSemver(semver.MustParse("0.5.3")),
+				},
+				FixedInVersion: "0.5.3",
+			},
+		},
+	},
+	{
+		name: "three_fixes",
+		ad: &advisory{
+			ID: "test1",
+			Affected: []affected{
+				{
+					Package: _package{
+						Ecosystem: "go",
+						Name:      "something",
+					},
+					Ranges: []_range{
+						{
+							Type: "SEMVER",
+							Events: []rangeEvent{
+								{
+									Introduced: "0",
+								},
+								{
+									Fixed: "2.1.16",
+								},
```

Comment:
is it also valid to not have `Introduced` here (I think no, but I think we cover that case? If we do, can you add another test for that case)?

Reply:
Added